cogboard-app/src/main/kotlin/com/cognifide/cogboard/logStorage/LogStorage.kt

```diff
@@ -0,0 +1,293 @@
+        val storageConfig = collectionConfiguration
+        var lastLine = storageConfig?.lastLine ?: 0
+        var seq = storageConfig?.seq ?: 0
+
+        // Get the number of lines in the file
+        val fileLineCount = connection.getNumberOfLines() ?: 0
+
+        if (fileLineCount > 0 && fileLineCount > lastLine) {
+            // Download new logs and append them
+            val inserted = downloadInsertLogs(seq, lastLine)
+            lastLine += inserted
+            seq += inserted
+        } else if (fileLineCount in 1 until lastLine) {
+            // Remove all logs and download from the beginning
+            deleteAllLogs()
+            seq = 0
+            val inserted = downloadInsertLogs(seq)
+            lastLine = inserted
+            seq += inserted
+        }
+
+        // Save the new configuration
+        saveConfiguration(LogCollectionConfiguration(config.id, lastLine, seq))
+    }
+
+    /** Returns all logs for this instance, sorted by their sequence number. */
+    private val logs: List<Log>
+    get() = logsCollection
+            ?.find()
+            ?.limit(config.logLines.toInt())
+            ?.sort(descending(Log.SEQ))
+            ?.mapNotNull { it }
+            ?.mapNotNull { Log.from(it) }
+            ?: emptyList()
+
+    /** Prepares a JSON response to be displayed */
+    private fun prepareResponse(): JsonObject {
+        return JsonObject(mapOf(
+            "variableFields" to parserStrategy.variableFields,
+            "logs" to JsonArray(logs.map { it.toJson() })
+        ))
+    }
+
+    /** Updates the logs and sends them to the widget. */
+    fun updateLogs(fetchNewLogs: Boolean) {
+        if (fetchNewLogs) {
+            // Download new logs
+            downloadLogs()
+
+            // Delete unnecessary logs
+            deleteOldLogs()
+            deleteSpaceConsumingLogs()
+        }
+
+        // Fetch the logs from the database and send them back
+        val response = prepareResponse()
+        vertx?.eventBus()?.send(config.eventBusAddress, response)
+    }
+
+    /** Deletes all data associated with the widget. */
+    fun delete() {
+        deleteAllLogs()
+        saveConfiguration(null)
+    }
+
+    companion object {
+        private const val DATABASE_NAME: String = "logs"
+        private const val CONFIGURATION_COLLECTION_NAME: String = "config"
+        private const val STATS_COMMAND: String = "collStats"
+        private const val STATS_SIZE: String = "size"
+        private const val MB_TO_BYTES: Long = 1024L * 1024L
+        private const val DAY_TO_TIMESTAMP = 24 * 60 * 60
+        private const val MONGO_SCHEME = "mongodb"
+        private val MONGO_USERNAME = System.getenv("MONGO_USERNAME") ?: "root"
+        private val MONGO_PASSWORD = System.getenv("MONGO_PASSWORD") ?: "root"
+        private val MONGO_HOST = System.getenv("MONGO_HOST") ?: "mongo"
+        private val MONGO_PORT = System.getenv("MONGO_PORT")?.toIntOrNull() ?: 27017
```

Comment:
Those need to be mentioned in the documentation of the widget

Reply:
Done.